parlai/mturk/core/react_server/dev/app.jsx

```diff
@@ -0,0 +1,668 @@
+      e.nativeEvent.stopImmediatePropagation();
+    }
+  }
+
+  render() {
+    // TODO maybe move to CSS?
+    let pane_style = {
+      'paddingLeft': '25px',
+      'paddingTop': '20px',
+      'paddingBottom': '20px',
+      'paddingRight': '25px',
+      'float': 'left',
+      'width': '100%'
+    };
+    let input_style = {
+      height: "50px", width: "100%", display: "block", float: 'left'
+    };
+    let submit_style = {
+      'width': '100px', 'height': '100%', 'fontSize': '16px',
+      'float': 'left', 'marginLeft': '10px', 'padding': '0px'
+    };
+
+    let text_input = (
+      <FormControl
+        type="text"
+        id="id_text_input"
+        style={{width: '80%', height: '100%', float: 'left', 'fontSize': '16px'}}
+        value={this.state.textval}
+        placeholder="Please enter here..."
+        onKeyPress={(e) => this.handleKeyPress(e)}
+        onChange={(e) => this.setState({textval: e.target.value})}
+        disabled={!this.props.active}/>
+    );
+
+    let submit_button = (
+      <Button
+        className="btn btn-primary"
+        style={submit_style}
+        id="id_send_msg_button"
+        disabled={this.state.textval == '' || !this.props.active}
+        onClick={() => this.tryMessageSend()}>
+          Send
+      </Button>
+    );
+
+    return (
+      <div
+        id="response-type-text-input"
+        className="response-type-module"
+        style={pane_style}>
+          <div style={input_style}>
+            {text_input}
+            {submit_button}
+          </div>
+      </div>
+    );
+  }
+}
+
+class ResponsePane extends React.Component {
+  render() {
+    let response_pane = null;
+    switch (this.props.chat_state) {
+      case 'done':
+      case 'inactive':
+        response_pane = <XDoneResponse
+          {...this.props}
+        />;
+        break;
+      case 'text_input':
+      case 'waiting':
+        response_pane = <XTextResponse
+          {...this.props}
+          active={this.props.chat_state == 'text_input'}
+        />;
+        break;
+      case 'idle':
+      default:
+        response_pane = <XIdleResponse />;
+        break;
+    }
+
+    return (
+      <div
+        id="right-bottom-pane"
+        style={{width: '100%', 'backgroundColor': '#eee'}}>
+        {response_pane}
+      </div>
+    );
+  }
+}
+
+class RightPane extends React.Component {
+  render () {
+    // TODO move to CSS
+    let right_pane = {
+      'minHeight': '100%', 'display': 'flex', 'flexDirection': 'column',
+      'justifyContent': 'spaceBetween'
+    };
+
+    return (
+      <div id="right-pane" style={right_pane}>
+        <XChatPane message_count={this.props.messages.length} {...this.props} />
+        <XResponsePane {...this.props} />
+      </div>
+    );
+  }
+}
+
+class LeftPane extends React.Component {
+  render () {
+    let frame_height = this.props.frame_height;
+    let frame_style = {
+      height: frame_height + 'px',
+      'backgroundColor': '#dff0d8',
```

Comment:
It is indeed - though I'm happy to change it hahaha. A followup is going to refactor a lot of the style into CSS so that feels like the more appropriate place to start making changes. Primary goal of this PR was to take the existing UI and port it directly to react.